src/python/pants/engine/internals/graph_test.py

```diff
@@ -844,6 +847,72 @@ class IrrelevantSources(Sources):
 # -----------------------------------------------------------------------------------------------
 
 
+def test_parse_dependencies_field() -> None:
+    result = parse_dependencies_field(
+        [
+            ":relative",
+            "//:top_level",
+            "demo:tgt",
+            "demo",
+            "./relative.txt",
+            "./child/f.txt",
+            "demo/f.txt",
+            "//top_level.txt",
+            "top_level2.txt",
+            # For files without an extension, you must use `./` There is no way (yet) to reference
+            # a file above you without a file extension.
+            "demo/no_extension",
+            "//demo/no_extension",
```

Comment:
NB that we cannot call `Path.is_file()` because that breaks the engine sandbox.

Reply:
Hm... that's actually not too bad a workaround? We should be able to loosen the restriction here later if need be, assuming that the error that you get when we think something is a target but it's actually a file is reasonable.

Reply:
To clarify, there is no workaround for a file that lives above the current BUILD file. The workaround only works for siblings and children.

But I suspect the edge case is unlikely enough to happen that this is okay. And there's a fallback to use an explicit target.